lib/BaseProjectLibrary.ts

```diff
@@ -119,6 +123,53 @@ export class BaseProjectLibrary implements ProjectLibrary {
 	public getComponentByName(name: string): Component {
 		return this.components.find(x => x.name === name);
 	}
+	public getComponentDescriptionByName(name: string): string {
+		return this.components.find(x => x.name === name).templates[0].description;
+	}
+
+	public getCustomTemplates(): Template[] {
+		return this.customTemplates;
+	}
+
+	public formatOutput(templates: Template[]): string[] {
```

Comment:
This one should be part of the prompt session logic (handling templates) and the general formatting should be in Util seeing as the `list` command should be using it too.